modules/tide_site_alert/src/Plugin/Block/TideSiteAlertBlock.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Drupal\tide_site_alert\Plugin\Block;
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\site_alert\Entity\SiteAlert;
+use Drupal\site_alert\Plugin\Block\SiteAlertBlock;
+
+/**
+ * Implements TideSiteAlertBlock class.
+ *
+ * @Block(
+ *   id = "tide_site_alert_block",
+ *   admin_label = @Translation("Tide Site Alert"),
+ * )
+ */
+class TideSiteAlertBlock extends SiteAlertBlock {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $build = parent::build();
+    foreach ($build as $key => $item) {
+      if (!is_numeric($key)) {
+        continue;
+      }
+      $value = NestedArray::getValue($build, ['#cache', 'tags', $key]);
+      if ($value && strpos($value, 'site_alert:') !== FALSE) {
+        preg_match('#[0-9]+$#', $value, $match);
+        $site_alert = SiteAlert::load(reset($match));
+        $build[$key]['#alert']['start'] = date('d/m/Y H:i:s', strtotime($site_alert->getStartTime()));
```

Comment:
I'm a bit of a stickler for DateTimeImmutable since you can trust the time you set on it hasn't changed.

Reply:
Hey @krakerag , thanks for the CR. 
It is always good to learn new things! 
I have made a further change trying to use `DateTimeImmutable` class. please take a look.